lib/integrations/externalcloudaudit/bootstrap.go

```diff
@@ -0,0 +1,405 @@
+	// Creates a new lifecycle configuration for the bucket or replaces an existing lifecycle configuration.
+	PutBucketLifecycleConfiguration(ctx context.Context, params *s3.PutBucketLifecycleConfigurationInput, optFns ...func(*s3.Options)) (*s3.PutBucketLifecycleConfigurationOutput, error)
+}
+
+// Creates a bucket with the given name in the given region.
+// Long Term Storage buckets have the following properties:
+// * Bucket versioning enabled
+// * Object locking enabled with Governance mode and default retention of 4 years
+// * Object ownership set to BucketOwnerEnforced
+// * Default SSE-S3 encryption
+func createLTSBucket(ctx context.Context, clt createBucketClient, bucketName string, region string) error {
+	err := createBucket(ctx, clt, bucketName, region, true)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	_, err = clt.PutObjectLockConfiguration(ctx, &s3.PutObjectLockConfigurationInput{
+		Bucket: &bucketName,
+		ObjectLockConfiguration: &s3types.ObjectLockConfiguration{
+			ObjectLockEnabled: s3types.ObjectLockEnabledEnabled,
+			Rule: &s3types.ObjectLockRule{
+				DefaultRetention: &s3types.DefaultRetention{
+					Years: defaultRetentionYears,
+					// Modification is prohibited without BypassGovernancePermission iam permission
+					Mode: s3types.ObjectLockRetentionModeGovernance,
+				},
+			},
+		},
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	return nil
+}
+
+// createTransientBucket is similar to createLTSBucket however object locking is not enabled and instead a lifecycle
+// policy is created that cleans up transient storage:
+// * Query results expire after 1 day
+// * Large payloads expire after 4 days
+// * DeleteMarkers, NonCurrentVersions and IncompleteMultipartUploads are also removed
+func createTransientBucket(ctx context.Context, clt createBucketClient, bucketName string, region string) error {
+	err := createBucket(ctx, clt, bucketName, region, false)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	_, err = clt.PutBucketLifecycleConfiguration(ctx, &s3.PutBucketLifecycleConfigurationInput{
+		Bucket: &bucketName,
+		LifecycleConfiguration: &s3types.BucketLifecycleConfiguration{
+			Rules: []s3types.LifecycleRule{
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireLargePayloads"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 4,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/large_payloads",
+					},
+				},
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireQueryResults"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 1,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/query_results",
```

Comment:
I'm currently just using `/results` as the prefix, we can change one or the other, doesn't really matter https://github.com/gravitational/teleport/blob/9f8530e8dbba76ef29a916b78bc504ef16470c1d/api/types/externalcloudaudit/externalcloudaudit.go#L115

Reply:
So we use query_results in cloud due to it sharing the bucket with large payloads. I think results makes more sense in this case but wasn't sure if we should keep it the same between the two environments.

Reply:
Updated in [258a3b8](https://github.com/gravitational/teleport/pull/33272/commits/258a3b815bd3c297c107036299d011e2549122a4)